Marlin/src/lcd/extui/ui_api.cpp

```diff
@@ -1025,9 +1029,13 @@ namespace ExtUI {
     TERN_(HAS_FAN, thermalManager.zero_fan_speeds());
   }
 
-  bool awaitingUserConfirm() { return TERN0(HAS_RESUME_CONTINUE, wait_for_user); }
+  bool awaitingUserConfirm() { return (TERN0(HAS_RESUME_CONTINUE, wait_for_user) || getMachineBusyState() >= 3); }
```

Comment:
If the busy-state enum values ever change, you can be sure we'll remember to change this "3" because I just wrote it down on a post-it note.

Reply:
The enum types didnt get passed through from the include for some reason, so i intend to chase it down tonight. Ill probably also add a storage function to hold the current state instead of just passing the message as well in case we want to call totally different screens or help control options shown from the onIdle.

Reply:
Wrapping it in `DEFER4()` should do the trick.

Reply:
Ah, nope… Just needs a `gcode.` prefix because the enum is a class member.

Reply:
Sounds good, because of the 2 bugfixes I wanted to get this up to see before I crashed last night but obviously want 100% done lol

Reply:
Close enough!